### ISSUE TITLE
DAOStarFinder throws an uncaught exception if no sources are detected

### ISSUE THREAD

**Issue by cmccully** [astropy/photutils#1510]
Photutils == 1.6.0 or main on github both produce this behavior.

```
from photutils import DAOStarFinder
import numpy as np
from scipy.ndimage import gaussian_filter
sigma = 1.0
data = np.zeros((100, 100))
err = np.ones_like(data)
xcoords = [10, 30, 50, 70]  # 4 features
ycoords = np.array(xcoords) + 5  # 4 features
data[ycoords, xcoords] = 1
data = gaussian_filter(data, sigma=sigma)
data /= np.max(data)
nsigma=2.
fwhm=6
daofind = DAOStarFinder(fwhm=fwhm, threshold=nsigma, exclude_border=True)
features = daofind(data / err, mask=np.zeros(data.shape, dtype=int))
```

produces the following error

```
File site-packages/photutils/detection/core.py:111, in StarFinderBase._find_stars(convolved_data, kernel, threshold, min_separation, mask, exclude_border)
    109     xmax = convolved_data.shape[1] - xpad
    110     ymax = convolved_data.shape[0] - ypad
--> 111     mask = ((tbl['x_peak'] > xpad) & (tbl['y_peak'] > ypad)
    112             & (tbl['x_peak'] < xmax) & (tbl['y_peak'] < ymax))
    113     tbl = tbl[mask]
    115 if tbl is None:

TypeError: 'NoneType' object is not subscriptable
```

**Comment by larrybradley** [comment]
@cmccully Many thanks for reporting this. It's fixed by #1512.

**Comment by cmccully** [comment]
Thanks @larrybradley!
